### PR TITLE
React-Redux Space Travel App 

### PR DESCRIPTION
**Pull request by sumairq**
# It is the pull request for the space travel group project, we implemented the following features:

##  Rockets Page

- Fetch rockets from the [spaceX rockets API endpoint](https://api.spacexdata.com/v3/rockets) and set the state with the data.
- Display rockets retrieved from the redux state.
- Reservation actions were implemented for making a reservation and canceling a reservation.
- Styling the rockets page. 

## Dragons Page

- Fetched Dragons data from the  [spaceX dragons API endpoint](https://api.spacexdata.com/v3/dragons) and set the state with the data.
- Display dragons retrieved from the redux state.
- Reservation actions were implemented for making a reservation and canceling a reservation.
- Styling the dragons page using react-bootstrap. 

## Missions Page

- Fetched Missions data from the  [spaceX missions API endpoint](https://api.spacexdata.com/v3/missions) and set the state with the data.
- Display missions retrieved from the redux state.
- Joining actions were implemented for joining a mission and leaving a mission.
- Styling the missions page using react-bootstrap. 

## Profile Page

- Combining the data from all the states to show the user's reservations from all pages.
- Filtered data from the state to only show the items that were reserved.
- Used react-bootstrap tables for housing the data like a dashboard.

## Testing

- Tests were written using the jest testing library.
- Tests included the interaction with the redux state.
- Mocks for the data were created
- We did snapshot testing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,113 @@
...
 
-In the project directory, you can run:
+![screenshot](Animation02.gif)
 
-### `npm start`
+## Live Demo
 
-Runs the app in the development mode.\
-Open [http://localhost:3000](http://localhost:3000) to view it in your browser.
+[![Netlify Status / heroku](xxxxxxxxxxxxxx/)
```
Comment (medaminedev66) [inline README.md:14]: - [ ] The `live link` and the `MIT link` in the README file are broken. I would highly recommend fixing those sections to make README professional.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi team, :wave:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

_See the inline comments below for more required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Dragons/Dragon.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import { useDispatch } from 'react-redux';
+import { PropTypes } from 'prop-types';
+import { addDragonReservation, removeDragonReservation } from '../../redux/dragons/dragons';
+
+const Dragon = (props) => {
+  const dispatch = useDispatch();
+  const {
+    id, name, image, desc, reservation, type,
+  } = props;
+  // console.log(desc);
```
Comment (medaminedev66) [inline src/components/Dragons/Dragon.js:15]: - [ ] Kindly remove the commented code so your code is clean.

File: src/components/Rockets/elementRockets.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { PropTypes } from 'prop-types';
+import styles from '../../pages/Rockets.module.css';
+import { addRocketReservation, removeRocketReservation } from '../../redux/rockets/rockets';
+
+const elementRockets = (props) => {
+  const dispatch = useDispatch();
+  const {
+    id, name, image, desc, reserved,
+  } = props;
+  // console.log(desc);
```
Comment (medaminedev66) [inline src/components/Rockets/elementRockets.js:12]: - [ ] Kindly remove the commented code so your code is clean.

File: src/components/Dragons/Dragon.js (hunk 2 above)
Comment (david-lafontant) [reply src/components/Dragons/Dragon.js]: Asked change implemented

File: src/components/Rockets/elementRockets.js (hunk 3 above)
Comment (david-lafontant) [reply src/components/Rockets/elementRockets.js]: Asked change implemented

File: README.md (hunk 1 above)
Comment (david-lafontant) [reply README.md:14]: MIT link is fixed

Comment (Whoistolu) [reply README.md]: All good now 👍🏽

File: src/components/Dragons/Dragon.js (hunk 2 above)
Comment (Whoistolu) [reply src/components/Dragons/Dragon.js]: All good now 👍🏽

File: src/components/Rockets/elementRockets.js (hunk 3 above)
Comment (Whoistolu) [reply src/components/Rockets/elementRockets.js]: All good now 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**